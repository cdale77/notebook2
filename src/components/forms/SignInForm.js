import React from "react";
import PropTypes from "prop-types";
import TextInput from "./TextInput";
import SubmitButton from "./SubmitButton";

class SignInForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = { email: "", password: "" };
  }

  handleEmailChange(e) {
    this.setState({ email: e.target.value });
  }

  handlePasswordChange(e) {
    this.setState({ password: e.target.value });
  }

  onSubmit(e) {
    e.preventDefault();
    this.props.onSubmit(this.state);
  }

  render() {
    return (
      <div className="sign-in-form">
        <form>
          <TextInput
            title="email"
            name="email"
            value={this.state.email}
            onChange={this.handleEmailChange.bind(this)}
          />
          <TextInput
            title="password"
            name="password"
            fieldType="password"
            value={this.state.password}
            onChange={this.handlePasswordChange.bind(this)}
          />
          <SubmitButton onSubmit={this.onSubmit.bind(this)} />
        </form>
      </div>
    );
  }
}

SignInForm.propTypes = {
  onSubmit: PropTypes.func.isRequired
};

export default SignInForm;
