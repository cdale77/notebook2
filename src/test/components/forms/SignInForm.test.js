import React from "react";
import Renderer from "react-test-renderer";
import TestUtils from "react-dom/test-utils";
import SignInForm from "../../../components/forms/SignInForm";

describe("SignInForm", () => {
  describe("rendering", () => {
    it("should render the form", () => {
      const component = Renderer.create(<SignInForm onSubmit={jest.fn()} />);

      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe("submitting the form", () => {
    let callBack, spy, component;

    beforeEach(() => {
      callBack = {
        onSubmit: () => {}
      };

      spy = jest.spyOn(callBack, "onSubmit");
      component = TestUtils.renderIntoDocument(
        <SignInForm onSubmit={callBack.onSubmit} />
      );
    });

    it("should call the submit function", () => {
      const button = TestUtils.findRenderedDOMComponentWithClass(
        component,
        "button"
      );

      const emailField = TestUtils.findRenderedDOMComponentWithClass(
        component,
        "input-email"
      );

      const pwField = TestUtils.findRenderedDOMComponentWithClass(
        component,
        "input-password"
      );

      // enter values in the fields to trigger the state change callbacks
      emailField.value = "changed";
      pwField.value = "password";
      TestUtils.Simulate.change(emailField);
      TestUtils.Simulate.change(pwField);

      TestUtils.Simulate.click(button);

      expect(spy).toHaveBeenCalled();
      expect(callBack.onSubmit.mock.calls.length).toEqual(1);

      // In the test env the password is coming as ""
      const form = { email: "changed", password: "" };
      expect(callBack.onSubmit.mock.calls).toEqual([[form]]);
    });
  });
});
