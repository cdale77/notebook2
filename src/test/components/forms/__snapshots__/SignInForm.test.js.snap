// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInForm rendering should render the form 1`] = `
<div
  className="sign-in-form"
>
  <form>
    <div
      className="form-group"
    >
      <input
        className="input-email"
        id="email"
        name="email"
        onChange={[Function]}
        placeholder="email"
        type="text"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <input
        className="input-password"
        id="password"
        name="password"
        onChange={[Function]}
        placeholder="password"
        type="password"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <input
        className="button"
        disabled={false}
        id="submit"
        name="submit"
        onClick={[Function]}
        type="submit"
        value="Submit"
      />
    </div>
  </form>
</div>
`;
